flight/mcu/lib/teensyShared/FSMCollection.h

```diff
@@ -1,19 +1,29 @@
 #pragma once
-#include <ChRt.h>
 
-#include "rocketFSM.h"
+#include "ChRt.h"
+#include <initializer_list>  // this is here to make initializing the FSMCollection a lot more convenient
+
+#include "RocketFSMBase.h"
 
 /**
  * @brief Manager for all FSMs in the rocket
  *
- * @param FSMs Array of pointers to all FSMs which implement RocketFSM
+ * @param FSMs Array of pointers to all FSMs which implement RocketFSMBase
  * @param count Number of FSMs in FSMs array
  */
 
 template <size_t count>
 class FSMCollection {
    public:
-    FSMCollection(RocketFSM** FSMs) : FSMs_(FSMs) {}
+    FSMCollection(std::initializer_list<RocketFSMBase*> fsms) {
+        size_t i = 0;
+        for (auto& item : fsms) {
+            if (i >= count) {
+                break;
+            }
+            FSMs_[i++] = item;
+        }
+    }
```

Comment:
Having the wrong number of fsms in the list should not be a silent error

Reply:
It's also in a constructor, so I can't really tell what to do here. Having a footgun here is actually just good C++ style.

Reply:
There might be a way to make this fail a static assert, if not i think stalling the program is the correct behavior here as this is a compile time issue and should be fixed